### ISSUE TITLE
.OrderBy() with dictionary value doesn't work

### ISSUE THREAD

**Issue by robert-skarzycki** [JasperFx/marten#1219]
Having that model (https://gist.github.com/robert-skarzycki/8a35ae9ac0d8cca1985cff0e022671b2) and data inserting code (https://gist.github.com/robert-skarzycki/6bef9198e9b963e22b673bc134302263), this query fails:
```
var res = session
                .Query<Car>().OrderBy(p => p.Attributes["petrol"].Value);
```
returns data unordered:
```
#       Name    Manuf   Petrol  Abs
0       car-1   VW      diesel  -
1       car-2   VW      petrol  yes
2       car-3   Audi    lpg     -
```
when I expect them to be ordered like this:
```
#       Name    Manuf   Petrol  Abs
0       car-1   VW      diesel  -
1       car-3   Audi    lpg     -
2       car-2   VW      petrol  yes
```
I've found dirty hack to get this, but I guess this is not an expected implementation:
```
var res = session
                .Query<Car>().Where(p => p.MatchesSql("1=1  order by CAST(d.data -> 'Attributes' -> 'petrol' -> 'Value' as jsonb)"));
```

**Comment by oskardudycz** [comment]
@robert-skarzycki I don't think that we'll find time to add support for that soon.

As a workaround you can add getter property like:

```
public string Petrol { get { return p.Attributes["petrol"].Value; } }
```
Then use it in order by.

It might be that during our [Linq Support Overhaul](https://github.com/JasperFx/marten/issues/1201) some extended support will be added, but I doubt that indexing dictionary + inner object support will get to the list.

@jeremydmiller  thoughts?

**Comment by robert-skarzycki** [comment]
@oskardudycz, the getter-workaround is no-go for me: I can't predict what keys are in the dictionary (in fact, if I could, I would make them regular props, instead of dictionary...). However, I'm especially not happy about this dirty hack with passing orderby clause into `MatchesSql`. Bascially it is SQL injection here. :) That might be a solution to allow passing sql string into custom order by extension method, instead injecting via `MatchesSql`.

**Comment by oskardudycz** [comment]
Passing custom sql seems to me as viable option, that could give us some flexibility. Not ideal, but at least would allow you and other users not supported cases. @jeremydmiller what are your thoughts on that?

**Comment by jeremydmiller** [comment]
The custom SQL is all we can do for now without explicitly coding support for this exact usage in the Linq provider.

**Comment by oskardudycz** [comment]
@robert-skarzycki I'll work on possibility to provide custom sql in order by.

**Comment by jeremydmiller** [comment]
@oskardudycz @robert-skarzycki I'm going to take this. I think I can get it to fall out from an intended fix for #744

**Comment by jeremydmiller** [comment]
Closed by https://github.com/JasperFx/marten/commit/315b24b808fea1acbfe38467811bb92e575f4ad3
